File: api/src/main/java/dev/obscuria/fragmentum/registry/DeferredEntity.java
package dev.obscuria.fragmentum.registry;

import net.minecraft.core.Holder;
import net.minecraft.world.entity.Entity;
import net.minecraft.world.entity.EntityType;

import java.util.function.Supplier;

@SuppressWarnings("unused")
public class DeferredEntity<T extends Entity> extends Deferred<EntityType<?>, EntityType<T>>
{
    public DeferredEntity(Supplier<Holder<EntityType<?>>> supplier)
    {
        super(supplier);
    }
}

File: api/src/main/kotlin/dev/obscuria/fragmentum/registry/DeferredEntity.kt
@file:Suppress("unused", "RedundantNullableReturnType")

package dev.obscuria.fragmentum.registry

import net.minecraft.core.Holder
import net.minecraft.world.entity.Entity
import net.minecraft.world.entity.EntityType

class DeferredEntity<T : Entity>(supplier: () -> Holder<EntityType<*>>) :
    Deferred<EntityType<*>, EntityType<T>>(supplier)